aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -63,19 +69,40 @@ module AptosFramework::Token {
         amount: u64,
     }
 
+    /// create collection event with creator address and collection name
+    struct CreateCollectionEvent has drop, store {
+        creator: address,
+        collection_name: ASCII::String,
+    }
+
+    /// token creation event id of token created
+    struct CreateTokenEvent has drop, store {
+        id: TokenId,
+    }
+
+    /// token creation event id of token created
+    struct CreateNonFungibleTokenEvent has drop, store {
+        id: TokenId,
+    }
+
     //
     // Core data structures for creating and maintaining tokens
     //
 
-    /// A token creator may have many tokens and many collections of tokens
+    /// Represent token collection meta-data owned by creator
+    /// A creator can have multiple collections
     struct Collections has key {
         collections: Table<ASCII::String, Collection>,
         token_data: Table<TokenId, TokenData>,
         burn_capabilities: Table<TokenId, BurnCapability>,
         mint_capabilities: Table<TokenId, MintCapability>,
+        create_collection_events: EventHandle<CreateCollectionEvent>,
+        create_token_events: EventHandle<CreateTokenEvent>,
+        create_non_fungible_token_events: EventHandle<CreateNonFungibleTokenEvent>,
+
     }
 
-    /// The source of Tokens, their collection!
+    /// Represent collection metadata
```

Comment:
Represents /the/ collection metadata

nit review for metadata vs meta-data

Reply:
updated to metadata